subprojects/plugin-development/src/integTest/groovy/org/gradle/plugin/devel/plugins/PrecompiledGroovyPluginsIntegrationTest.groovy

```diff
@@ -371,9 +367,41 @@ class PrecompiledGroovyPluginsIntegrationTest extends AbstractIntegrationSpec {
 
         then:
         succeeds('help')
+        outputContains('pluginManagement block executed')
         outputContains('my-settings-plugin applied!')
     }
 
+    def "precompiled project plugin can not use buildscript block"() {
+        when:
+        enablePrecompiledPluginsInBuildSrc()
+        file("buildSrc/src/main/groovy/plugins/foo.gradle") << """
+            buildscript {
+                dependencies {}
+            }
+            println("foo project plugin applied")
+        """
+
+        then:
+        fails("help")
+        failureCauseContains("The `buildscript` block is not supported in Groovy script plugins, please use the `plugins` block or project level dependencies.")
+    }
+
+    @ToBeFixedForConfigurationCache(because = "groovy precompiled scripts")
+    def "precompiled settings plugin can not use buildscript block"() {
+        when:
+        enablePrecompiledPluginsInBuildSrc()
+        file("buildSrc/src/main/groovy/plugins/foo.settings.gradle") << """
+            buildscript {
+                dependencies {}
+            }
+            println("foo settings plugin applied")
+        """
+
+        then:
+        fails('help')
+        failureCauseContains("The `buildscript` block is not supported in Groovy script plugins, please use the `plugins` block or project level dependencies.")
+    }
```

Comment:
Those two make sure the compilation fails with a message that is similar to one produced by the Kotlin counterpart.